### PR TITLE
fix a typo in flax T5 attention - attention_mask variable is misnamed

### LINKED ISSUE #26564: Flax T5 model - code typo during AutoRegressive decoding?
### System Info

- `transformers` version: 4.33.3
- Platform: Linux-5.15.120+-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.17.3
- Safetensors version: 0.3.3
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.0.1+cu118 (False)
- Tensorflow version (GPU?): 2.13.0 (False)
- Flax version (CPU?/GPU?/TPU?): 0.7.4 (cpu)
- Jax version: 0.4.10
- JaxLib version: 0.4.10
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

@sanchit-gandhi

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

The discussed file is transformers/src/transformers/models/t5/modeling_flax_t5.py at line 408

https://github.com/huggingface/transformers/blob/2aef9a96011133f6b399b598fd69cfeca936eb37/src/transformers/models/t5/modeling_flax_t5.py#L408C1-L409C1

### Expected behavior

Hi,

During autoregressive decoding, keys and values are computed one token at a time and cache is used to recover the keys and values from previous calls.

In the `_concatenate_to_cache` method of the Attention module, an attention mask is computed in order for the new query to only attend to the previous key positions and not the remaining zero elements. This is what is explained in the comments in this function.

However, the new attention mask is not used afterwards, because its name is `attention_attention_mask` and not `attention_mask` which is the one being used in every other line. 

From my understanding, this is likely a typo and I am not sure how it changes the behavior of the model, if at all.

### PR DESCRIPTION
**Pull request by giganttheo**
# What does this PR do?

Fixes a typo in the Flax code for T5 model.

There is a typo in the Attention module of the Flax version of T5, where the attention_mask updated by the `_concatenate_to_cache` method should override the previous attention_mask but does not because of a misnamed variable.

Fixes #26564 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

@sanchit-gandhi

### CODE REVIEW

**Review by sanchit-gandhi — APPROVED** [review]
Very nice @giganttheo! Thanks for identifying the bug and proposing the fix 🤗 Confirming that the slow tests pass following the fix? As per https://github.com/huggingface/transformers/issues/26564#issuecomment-1751779825 If so, then this all LGTM!

**Comment by giganttheo** [comment]
> Very nice @giganttheo! Thanks for identifying the bug and proposing the fix 🤗 Confirming that the slow tests pass following the fix? As per [#26564 (comment)](https://github.com/huggingface/transformers/issues/26564#issuecomment-1751779825) If so, then this all LGTM!

The slow tests are passing for t5 and longt5:

`RUN_SLOW=1 pytest -sv tests/models/t5/test_modeling_flax_t5.py::FlaxT5ModelIntegrationTests`

outputs: `================== 6 passed, 4 warnings in 331.38s (0:05:31) ===================`

and for the longT5 version:

`RUN_SLOW=1 pytest -sv tests/models/longt5/test_modeling_flax_longt5.py::FlaxLongT5ModelIntegrationTests`

outputs: `=================== 1 passed, 1 warning in 401.61s (0:06:41) ===================`

**Review by ArthurZucker — APPROVED** [review]
LGTM! Thanks for catching 🤗
